src/Loop.php

```diff
@@ -42,12 +42,11 @@ public static function setFactory(DriverFactory $factory = null)
         }
 
         self::$factory = $factory;
+        self::$driver = null; // reset it here, it will be actually instantiated inside execute() or get()
 
         if ($factory === null) {
-            self::$driver = null;
             self::$registry = null;
         } else {
-            self::$driver = self::createDriver();
             self::$registry = [];
```

Comment:
There may be reasons to reuse the loop. I.e. an exception was thrown and you handled it outside. The loop shall be re-runnable then.

Making it a trait for usage in Drivers is an option. Not sure if the best one, but it isn't bad.